packages/react/src/components/Tab/Tab.js

```diff
@@ -122,6 +127,7 @@ export default class Tab extends React.Component {
 
   render() {
     const {
+      id,
```

Comment:
Should we auto-generate this id if they don't supply one?

Reply:
Agreed that auto-generation is desirable, but couldn't see a way for [`<TabContent>` generation code](https://github.com/carbon-design-system/carbon/blob/v10.9.0/packages/react/src/components/Tabs/Tabs.js#L236-L248) to get such auto-generated ID.

Reply:
Definitely understand, this is where some of our context experiments have taken us to help coordinate registering these ids.

Reply:
Right that we need some improvements in this area, I expect other approaches like render-props take into account for that.